parser/structured_data.go

```diff
@@ -191,29 +204,39 @@ func structuredDataParser(data []byte, removeEscapedCharsFromResult bool) (int,
 			continue
 		}
 
-		if c == ' ' {
+		// Stop when we reach a space, unless we're
+		// instructed to only stop on new metrics,
+		// in which case we will keep going until
+		// we find a closing bracket.
+		if !stopOnNewMetric && c == ' ' {
 			break
 		}
 	}
 
-	skippedWidth := 0
+	// Prepare the return values
+	data = bytes[:start]
```

Comment:
This is a new way of looking at the word "start" :D

Reply:
Yeah... I started refactoring the names there but I obviously _start_ed in the wrong end... or stopped...